Cubical/Categories/DistLatticeSheaf/Extension.agda

```diff
@@ -0,0 +1,1011 @@
+              f ⋆⟨ C ⟩ coneOut F[⋁α]Cone ((u ∧l α i , ∧lClosed _ _ u∈L' (α∈L' i)) , u∧αᵢ≤⋁α i)
+            ≡⟨ cong (λ x → f ⋆⟨ C ⟩ x) (sym (coneOutCommutes F[⋁α]Cone
+                    (≤m→≤j _ _ (∧≤LCancel _ _) , is-prop-valued _ _ _ _))) ⟩
+              f ⋆⟨ C ⟩ (coneOut F[⋁α]Cone ((α i , α∈L' i) , ind≤⋁ α i)
+                ⋆⟨ C ⟩ F .F-hom (≤m→≤j _ _ (∧≤LCancel _ _)))
+            ≡⟨ sym (⋆Assoc C _ _ _) ⟩
+              (f ⋆⟨ C ⟩ coneOut F[⋁α]Cone ((α i , α∈L' i) , ind≤⋁ α i))
+                 ⋆⟨ C ⟩ F .F-hom (≤m→≤j _ _ (∧≤LCancel _ _))
+            ≡⟨ cong (λ x → x ⋆⟨ C ⟩ F .F-hom (≤m→≤j _ _ (∧≤LCancel _ _))) (isRestConeMorf (sing i)) ⟩
+              coneOut cc (sing i) ⋆⟨ C ⟩ F .F-hom (≤m→≤j _ _ (∧≤LCancel _ _)) ∎
+
+    -- the other direction, surprisingly hard
+    lemma3 : ∀ (c : ob C) (cc : Cone (funcComp F (BDiag (λ i → α i , α∈L' i))) c)
+               (f : C [ c , DLRan F .F-ob (⋁ α) ])
+           → isConeMor (lemma1 c cc) F[⋁α]Cone f
+           → isConeMor cc restCone f
+    lemma3 c cc f isConeMorF = isConeMorSingLemma _ cc restCone singCase
+      where
+      singCase : ∀ i → f ⋆⟨ C ⟩ coneOut restCone (sing i) ≡ coneOut cc (sing i)
+      singCase i =
+        (subst (λ g → f ⋆⟨ C ⟩ (F[⋁α]Cone .coneOut ((α i , α∈L' i) , ind≤⋁ α i)) ≡ g)
+          (transport (λ j → helperPathP j ≡ ccᵢSubstFiller (~ j)) ccᵢSubstPath)
+            assumption)
+        where
+        assumption : f ⋆⟨ C ⟩ (F[⋁α]Cone .coneOut ((α i , α∈L' i) , ind≤⋁ α i))
+                   ≡ coneOut (lemma1 c cc) ((α i , α∈L' i) , ind≤⋁ α i)
+        assumption = isConeMorF ((α i , α∈L' i) , ind≤⋁ α i)
+
+        -- modulo transport
+        Σpathhelper : (α i , α∈L' i) ≡ (⋁ (β (α i)) , ⋁β∈L' (α i) (α∈L' i) (ind≤⋁ α i))
+        Σpathhelper = Σ≡Prop (λ x → L' x .snd) (β≡ (α i) (ind≤⋁ α i))
+
+        Σpathhelper2 : (⋁ (β (α i)) , ⋁β∈L' (α i) (α∈L' i) (ind≤⋁ α i)) ≡ (α i , α∈L' i)
+        Σpathhelper2 = Σ≡Prop (λ x → L' x .snd) (sym (β≡ (α i) (ind≤⋁ α i)))
+
+        ccᵢSubst : C [ c , F .F-ob  (⋁ (β (α i)) , ⋁β∈L' (α i) (α∈L' i) (ind≤⋁ α i)) ]
+        ccᵢSubst = subst (λ x → C [ c , F .F-ob x ])
+                     (Σ≡Prop (λ x → L' x .snd) (β≡ (α i) (ind≤⋁ α i)))
+                     (coneOut cc (sing i))
+
+        ccᵢSubstFiller : PathP (λ j → C [ c , F .F-ob (Σpathhelper j) ])
+                               (coneOut cc (sing i)) ccᵢSubst
+        ccᵢSubstFiller = subst-filler (λ x → C [ c , F .F-ob x ]) Σpathhelper (coneOut cc (sing i))
+
+        βSubstFiller : PathP (λ j → C [ c , F .F-ob (Σpathhelper2 j) ])
+                    (uniqβConeMor c cc (α i) (α∈L' i) (ind≤⋁ α i) .fst .fst)
+                    (coneOut (lemma1 c cc) ((α i , α∈L' i) , ind≤⋁ α i))
+        βSubstFiller = subst-filler (λ x → C [ c , F .F-ob x ]) Σpathhelper2
+                                    (uniqβConeMor c cc (α i) (α∈L' i) (ind≤⋁ α i) .fst .fst)
+
+        Σpathhelperpath : Σpathhelper2 ≡ sym Σpathhelper
```

Comment:
?

Reply:
The problem is that they're not equal by refl, but I have an idea to drastically reduce transports.